src/python-bindings/classad.cpp

```diff
@@ -783,7 +783,8 @@ pythonFunctionTrampoline_internal(const char *name, const classad::ArgumentList&
         pyKw["state"] = wrapper;
     }
 
-    boost::python::object pyResult = py_import("__main__").attr("__builtins__").attr("apply")(pyFunc, pyArgs, pyKw);
+    boost::python::object pyResult = boost::python::eval("lambda f,a,kw: f(*a,**kw)")(pyFunc, pyArgs, pyKw);
```

Comment:
Sure.  Would you like to leave it a single code path (as it currently is in this PR) or would you prefer to conditionalize it and leave the old "apply" behavior for python 2?